factcast-site/documentation/content/factus/spring_transaction.md

```diff
@@ -0,0 +1,268 @@
+                "SELECT fact_position FROM fact_positions WHERE projection_name = ?",
+                UUID.class,
+                getScopedName().asString());
+    } catch (IncorrectResultSizeDataAccessException e) {
+        // no state yet, just return null
+        return null;
+    }
+}
+``` 
+
+In case no previous Fact position exists, `null` is returned. 
+
+
+### Applying Events
+
+As explained [in the introduction to transactional projections]({{< ref "transactional_projections.md" >}}),
+when receiving an event, two write operations towards the data-store (in our case a JDBC database) happen:
+
+1. the projection is updated
+2. the Fact stream position is persisted
+
+The latter step was already discussed in the previous sections. 
+What is left, is to actually handle the received events and update the projection.
+
+As the concrete structure of your database tables depend on your use-case, here is an example projection which
+handles *UserCreated* and *UserDeleted* events. The projection data is persisted in the table `users`:
+
+```sql
+CREATE TABLE users (
+    name TEXT, 
+    id UUID, 
+    PRIMARY KEY (id));
+```
+
+When processing the *UserCreated* event, we add a new row to the `users` tables, filled with event data: 
+
+```java
+@Handler
+void apply(UserCreated e) {
+    jdbcTemplate.update(
+            "INSERT INTO users (name, id) VALUES (?,?);", 
+            e.getUserName(), 
+            e.getAggregateId());
+}
+```
+
+When handling the *UserDeleted* event we do the opposite and remove the appropriate row:
+
+```java
+@Handler
+void apply(UserDeleted e) {
+    jdbcTemplate.update("DELETE FROM users where id = ?", e.getAggregateId());
+}
+``` 
+
+We have finished the implementation of the event-processing part of our projection. What is missing is a way to 
+make the projection's data accessible for users.  
+
+
+### Providing Access To The Projection
+
+Users of our projections (meaning "other code") contact the projection via it's public API. 
+Currently, there is no public method offering "user names". So let's change that:
+
+```java
+public List<String> getUserNames() {
+    return jdbcTemplate.query("SELECT name FROM users", (rs, rowNum) -> rs.getString(1));
+}
+```
+
+Once again, we employ the `JdbcTemplate` to provide the list of user names. Since our projection data is stored in 
+a database, we simply use SQL to query the data.  
+
+As the last step, we now change the perspective and look at how other code can use our `UserNames` projection.
+
+
+### Using The Projection
+
+Calling code which wants to talk to the projection, requires only a few changes: 
+
+```java
+// not needed when the projection is a Spring @Component
+UserNames userNames = new UserNames(platformTransactionManager, jdbcTemplate);
+factus.update(userNames);
+List<String> userNames = uut.getUserNames();
+```
+
+First we create an instance of the projection and provide it with all required dependencies. Note, that in a Spring
```

Comment:
Decided to leave `@Component` out of the example above as I wanted to focus on transactionality and not Spring DI. However, somewhere I need to mention that you usually don't instantiate the projection yourself so there it is ;-)